File: cs/BootCampCode/testcases/chapterSix/CreateAccountTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;

namespace BootCampCode.testcases.chapterSix
{
    [TestClass]
    public class CreateAccountTest
    {
        [TestMethod]
        public void CreateAccount()
        {
            IWebDriver driver = new ChromeDriver();

            WebDriverWait myWaitVar = new WebDriverWait(driver, TimeSpan.FromSeconds(20));

            // Open the website
            driver.Navigate().GoToUrl("https://techblog.polteq.com/testshop/index.php");

            driver.Manage().Window.Maximize();

            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));

            driver.FindElement(By.ClassName("login")).Click();

            Assert.AreEqual("Authentication - TestShop", driver.Title);

            IWebElement createEmail = wait.Until(ExpectedConditions.ElementToBeClickable(By.Id("email_create")));
            createEmail.Click();

            Random random = new Random();

            int emailAddition = random.Next(1, 10000);

            string emailBase = "bootcampmania";
            string emailProvider = "@test.com";
            createEmail.SendKeys(emailBase + emailAddition + emailProvider);

            IWebElement createAccount = driver.FindElement(By.Id("SubmitCreate"));
            createAccount.Click();

            IWebElement firstGender = wait.Until(ExpectedConditions.ElementToBeClickable(By.Id("uniform-id_gender2")));
            firstGender.Click();

            IWebElement fName = driver.FindElement(By.Id("customer_firstname"));
            fName.SendKeys("T");

            IWebElement lName = driver.FindElement(By.Id("customer_lastname"));
            lName.SendKeys("Test");

            IWebElement password = driver.FindElement(By.Id("passwd"));
            password.SendKeys("1qazxsw2");

            var daySelection = driver.FindElement(By.Id("days"));
            var selectDay = new SelectElement(element: daySelection);
            selectDay.SelectByValue("1");

            var monthSelection = driver.FindElement(By.Id("months"));
            var selectMonth = new SelectElement(element: monthSelection);
            selectMonth.SelectByValue("1");

            var yearSelection = driver.FindElement(By.Id("years"));
            var selectYear = new SelectElement(element: yearSelection);
            selectYear.SelectByValue("1982");

            IWebElement submitButton = driver.FindElement(By.Id("submitAccount"));
            Console.WriteLine(submitButton.Text);
            submitButton.Click();
            
            // Verify if the MY ACCOUNT text is show
            String validationString = driver.FindElement(By.CssSelector("h1.page-heading")).Text;
            Assert.AreEqual("MY ACCOUNT", validationString, "My account element was not found");

            driver.Quit();
        }
    }
}

File: java/src/test/java/testcases/chapterSix/CreateAccountTest.java
package testcases.chapterSix;

import org.assertj.core.api.Assertions;
import org.openqa.selenium.By;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.Test;
import testcases.TestShopScenario;
import org.openqa.selenium.support.ui.Select;

import java.util.UUID;

public class CreateAccountTest extends TestShopScenario {

    @Test
    public void createAccount(){
        String email = UUID.randomUUID() + "@foobar.com";
        email = email.replace("-", "");
        WebDriverWait wait = new WebDriverWait(driver,20);
        driver.manage().window().maximize();
        driver.findElement(By.cssSelector("a.login")).click();

        driver.findElement(By.id("email_create")).sendKeys(email);
        driver.findElement(By.id("SubmitCreate")).click();

        wait.until(ExpectedConditions.elementToBeClickable(By.id("id_gender2"))).click();
        //driver.findElement(By.id("id_gender1")).click();
        driver.findElement(By.id("customer_firstname")).sendKeys("Tester");
        driver.findElement(By.id("customer_lastname")).sendKeys("QA");
        driver.findElement(By.id("passwd")).sendKeys("1qazxsw2");

        new Select(driver.findElement(By.id("days"))).selectByValue("25");
        new Select(driver.findElement(By.id("months"))).selectByValue("10");
        new Select(driver.findElement(By.id("years"))).selectByValue("1971");

        driver.findElement(By.id("newsletter")).click();
        driver.findElement(By.id("optin")).click();
        driver.findElement(By.id("submitAccount")).click();

        Assertions.assertThat(driver.findElement(By.cssSelector("a.account > span")).getText())
                .as("Username in header").isEqualTo("Tester QA");
        Assertions.assertThat(driver.findElement(By.cssSelector("p.alert.alert-success")).getText())
                .as("Message that I have created an account").isEqualTo("Your account has been created.");
        Assertions.assertThat(driver.findElement(By.tagName("h1")).getText())
                .as("Page Title of My Account Page").isEqualTo("MY ACCOUNT");
        Assertions.assertThat(driver.getTitle())
                .as("Browser title").contains("My account");
    }
}
